Subject: [SCSI] ipr: qc_fill_rtf() method should not store alternate status register
From: Sergei Shtylyov <sergei.shtylyov@cogentembedded.com>
Git-commit: 32d3d793fb2d40d2a255af64ffa17cfb2132c323
Patch-mainline: v3.10-rc3-64-g32d3d79
References: fate#317607

The 'ctl' field of  the 'struct ata_taskfile' is not really dual purpose, i.e.
it is not intended  for storing the alternate status register (which is mapped
at the same address in the legacy IDE controllers) in the qc_fill_rtf() method.
No other 'libata' driver except 'drivers/scsi/ipr.c' stores the alternate status
register's value in the 'ctl' field of 'qc->result_tf', hence this driver should
not do this as well...

Signed-off-by: Sergei Shtylyov <sergei.shtylyov@cogentembedded.com>
Acked-by: Brian King <brking@linux.vnet.ibm.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Acked-by: Dinar Valeev <dvaleev@suse.com>

---
 drivers/scsi/ipr.c |    1 -
 1 file changed, 1 deletion(-)

Index: b/drivers/scsi/ipr.c
===================================================================
--- a/drivers/scsi/ipr.c	2014-12-10 02:27:13.171742663 -0600
+++ b/drivers/scsi/ipr.c	2014-12-10 02:28:10.131743001 -0600
@@ -6659,7 +6659,6 @@ static bool ipr_qc_fill_rtf(struct ata_q
 	tf->hob_lbal = g->hob_lbal;
 	tf->hob_lbam = g->hob_lbam;
 	tf->hob_lbah = g->hob_lbah;
-	tf->ctl = g->alt_status;
 
 	return true;
 }
